lib/ts/recipe/session/recipe.ts

```diff
@@ -19,18 +19,27 @@
 import RecipeModule from "../recipeModule";
 import { CreateRecipeFunction, NormalisedAppInfo, RecipeFeatureComponentMap } from "../../types";
 import { isTest } from "../../utils";
-import { InputType } from "./types";
+import { InputType, OnHandleEventContext } from "./types";
 import sessionSdk from "supertokens-website";
 
+type ConfigType = InputType & { recipeId: string; appInfo: NormalisedAppInfo };
+
 export default class Session extends RecipeModule<unknown, unknown, unknown, any> {
     static instance?: Session;
     static RECIPE_ID = "session";
 
-    constructor(config: InputType & { recipeId: string; appInfo: NormalisedAppInfo }) {
+    private eventListeners = new Set<(ctx: OnHandleEventContext) => void>();
+
+    constructor(config: ConfigType) {
         super(config);
 
+        if (config.onHandleEvent !== undefined) {
+            this.addEventListener(config.onHandleEvent);
+        }
+
         sessionSdk.init({
             ...config,
+            onHandleEvent: this.onHandleEvent,
```

Comment:
We need to call the user's `onHandleEvent` as well?

Reply:
Not exactly. The handler itself will be called by this simplified listener/observer code https://github.com/supertokens/supertokens-auth-react/pull/274/files#diff-dd812fd78904a37e7eb4dc6156f8e62e23cf966041cfc5d89120378b0dd5e9d6R101

We add the user's event handler here https://github.com/supertokens/supertokens-auth-react/pull/274/files/1fec5f5621ab9d9ccdfedf6c0c65055d3fe48d1f#diff-dd812fd78904a37e7eb4dc6156f8e62e23cf966041cfc5d89120378b0dd5e9d6R36

So `Session` RecipeModule will call all the listeners that we have in `eventListeners` set. This includes user event listener and `SessionAuth` event listener.

Reply:
So user's `onHandleEvent` will be called, but not by decorating the function.

Reply:
Sounds good. Sorry, missed that line of code whilst seeing it...